spec/lib/hca/enrollment_system_spec.rb

```diff
@@ -1579,10 +1579,26 @@
     end
 
     context 'with attachments' do
-      it 'creates the right result', run_at: '2019-01-11 14:19:04 -0800' do
-        health_care_application = build(:hca_app_with_attachment)
-        result = described_class.veteran_to_save_submit_form(health_care_application.parsed_form, nil, '10-10EZ')
-        expect(result.to_json).to eq(get_fixture('hca/result_with_attachment').to_json)
+      context 'HCA attachment' do
+        it 'creates the right result', run_at: '2019-01-11 14:19:04 -0800' do
+          health_care_application = build(:hca_app_with_attachment)
+          result = described_class.veteran_to_save_submit_form(health_care_application.parsed_form, nil, '10-10EZ')
+          expect(result.to_json).to eq(get_fixture('hca/result_with_attachment').to_json)
+        end
+      end
+
+      context 'Form1010Ezr attachment' do
+        it 'creates the right result', run_at: '2024-06-27 18:22:17 -0800' do
+          parsed_form = get_fixture('form1010_ezr/valid_form').merge(
+            'attachments' => [
+              {
+                'confirmationCode' => create(:form1010_ezr_attachment).guid
+              }
+            ]
+          )
+          result = described_class.veteran_to_save_submit_form(parsed_form, nil, '10-10EZR')
+          expect(result.to_json).to eq(get_fixture('form1010_ezr/result_with_attachment').to_json)
+        end
```

Comment:
Added a test to confirm that ezr attachments are being added by the `save_to_submit_form` method.

Reply:
Thanks for adding this! I wonder if we shouldn't handle for the case where neither an `HCAAttachment` or `Form1010EzrAttachment` is found, resulting in the `form_attachment` being `nil`.  At the moment if neither was found this would just blow up with the error `NoMethodError: undefined method `get_file' for nil:NilClass` since the  `form_attachment` is nil. Probably very much an edge case though 🤷 .

Reply:
I added a `next` call to the code and wrote a new test to cover that scenario! 😄